openedx/features/course_experience/utils.py

```diff
@@ -125,6 +125,22 @@ def mark_last_accessed(user, course_key, block):
                 # we'll use the last child.
                 block['children'][-1]['resume_block'] = True
 
+    def recurse_mark_scored(block):
+        """
+        Mark this block as 'scored' if any of its descendents are 'scored' (that is, 'has_score' and 'weight' > 0).
+        """
+        is_scored = block.get('has_score', False) and block.get('weight', 1) > 0
```

Comment:
is `1` the correct default for `weight`?  So, should a None block weight result in is_scored being True?

Reply:
Yeah. https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/capa_base.py#L394-L405 is the relevant code.